ignite/contrib/metrics/precision_recall_curve.py

```diff
@@ -1,19 +1,22 @@
-from typing import Any, Callable, Tuple
+from typing import Any, Callable, Tuple, Union
 
 import torch
 
 from ignite.metrics import EpochMetric
 
 
-def precision_recall_curve_compute_fn(y_preds: torch.Tensor, y_targets: torch.Tensor) -> Tuple[Any, Any, Any]:
-    try:
-        from sklearn.metrics import precision_recall_curve
-    except ImportError:
-        raise RuntimeError("This contrib module requires sklearn to be installed.")
+def precision_recall_curve_compute_fn(y_preds: torch.Tensor, y_targets: torch.Tensor) -> Union[Any, Any, Any]:
+    from sklearn.metrics import precision_recall_curve
 
-    y_true = y_targets.numpy()
-    y_pred = y_preds.numpy()
-    return precision_recall_curve(y_true, y_pred)
+    y_true = y_targets.cpu().numpy()
+    y_pred = y_preds.cpu().numpy()
+    precision, recall, thresholds = precision_recall_curve(y_true, y_pred)
+    precision, recall, thresholds = (
+        torch.from_numpy(precision.copy()),
```

Comment:
You mention negative strides here in numpy array. This is unusual I'd say... Can't make those arrays contiguous (which will certainly copy the data)?

Reply:
Yes i can use `numpy.asascontiguousarray` before converting to tensors, this will work, what about that?

Reply:
Actually, can you paste here a code to repro the output with negative strides in numpy array ?

Reply:
```
y = np.random.randint(0, 2, size=(5,1))
y_pred = np.random.rand(5,1)
np_precision, np_recall, np_thresholds = precision_recall_curve(y, y_pred)
precision, recall, thresholds = torch.from_numpy(np_precision), torch.from_numpy(np_recall), torch.from_numpy(np_thresholds)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: At least one stride in the given numpy array is negative, and tensors with negative strides are not currently supported. (You can probably work around this by making a copy of your array  with array.copy().) 
```
The problem here is with `thresholds`
```
precision, recall, thresholds = torch.from_numpy(np_precision), torch.from_numpy(np_recall), torch.from_numpy(np_thresholds.copy())
```
This works fine

Reply:
Sounds to copy thresholds. Please, also put a comment about negatives stride in thresholds and a workaround by `.copy`.